pkg/machinery/config/types/v1alpha1/v1alpha1_provider_clusterconfig.go

```diff
@@ -0,0 +1,255 @@
+func (c *ClusterConfig) ExtraManifestURLs() []string {
+	return c.ExtraManifests
+}
+
+// ExtraManifestHeaderMap implements the config.ClusterConfig interface.
+func (c *ClusterConfig) ExtraManifestHeaderMap() map[string]string {
+	return c.ExtraManifestHeaders
+}
+
+// AdminKubeconfig implements the config.ClusterConfig interface.
+func (c *ClusterConfig) AdminKubeconfig() config.AdminKubeconfig {
+	return c.AdminKubeconfigConfig
+}
+
+// ScheduleOnMasters implements the config.ClusterConfig interface.
+func (c *ClusterConfig) ScheduleOnMasters() bool {
+	return c.AllowSchedulingOnMasters
+}
+
+// config.Token methods.
+
+// ID implements the config.Token interface.
+func (c *ClusterConfig) ID() string {
+	parts := strings.Split(c.BootstrapToken, ".")
+	if len(parts) != 2 {
+		return ""
+	}
+
+	return parts[0]
+}
+
+// Secret implements the config.Token interface.
+func (c *ClusterConfig) Secret() string {
+	parts := strings.Split(c.BootstrapToken, ".")
+	if len(parts) != 2 {
+		return ""
+	}
+
+	return parts[1]
+}
+
+// config.ClusterNetwork methods.
+
+// CNI implements the config.ClusterNetwork interface.
+func (c *ClusterConfig) CNI() config.CNI {
+	switch {
+	case c.ClusterNetwork == nil:
+		fallthrough
+
+	case c.ClusterNetwork.CNI == nil:
+		return &CNIConfig{
+			CNIName: constants.DefaultCNI,
+		}
+	}
+
+	return c.ClusterNetwork.CNI
+}
+
+// PodCIDR implements the config.ClusterNetwork interface.
+func (c *ClusterConfig) PodCIDR() string {
+	switch {
+	case c.ClusterNetwork == nil:
+		fallthrough
+	case len(c.ClusterNetwork.PodSubnet) == 0:
+		return constants.DefaultIPv4PodNet
+	}
+
+	return strings.Join(c.ClusterNetwork.PodSubnet, ",")
+}
+
+// ServiceCIDR implements the config.ClusterNetwork interface.
+func (c *ClusterConfig) ServiceCIDR() string {
+	switch {
+	case c.ClusterNetwork == nil:
+		fallthrough
+	case len(c.ClusterNetwork.ServiceSubnet) == 0:
+		return constants.DefaultIPv4ServiceNet
+	}
+
+	return strings.Join(c.ClusterNetwork.ServiceSubnet, ",")
+}
+
+// DNSDomain implements the config.ClusterNetwork interface.
+func (c *ClusterConfig) DNSDomain() string {
+	if c.ClusterNetwork == nil {
+		return constants.DefaultDNSDomain
+	}
+
+	return c.ClusterNetwork.DNSDomain
+}
+
+// APIServerIPs implements the config.ClusterNetwork interface.
+func (c *ClusterConfig) APIServerIPs() ([]net.IP, error) {
+	serviceCIDRs, err := talosnet.SplitCIDRs(c.ServiceCIDR())
+	if err != nil {
+		return nil, fmt.Errorf("failed to process Service CIDRs: %w", err)
+	}
+
+	return talosnet.NthIPInCIDRSet(serviceCIDRs, 1)
+}
+
+// DNSServiceIPs implements the config.ClusterNetwork interface.
+func (c *ClusterConfig) DNSServiceIPs() ([]net.IP, error) {
+	serviceCIDRs, err := talosnet.SplitCIDRs(c.ServiceCIDR())
+	if err != nil {
+		return nil, fmt.Errorf("failed to process Service CIDRs: %w", err)
+	}
+
+	return talosnet.NthIPInCIDRSet(serviceCIDRs, 10)
+}
+
+// Check interfaces.
+var (
+	_ config.ClusterConfig  = (*ClusterConfig)(nil)
```

Comment:
nit: we ended up using unit-tests for such checks, `assert.Implements` from stretchr/testify